presto-spark-base/src/main/java/com/facebook/presto/spark/execution/property/NativeExecutionSystemConfig.java

```diff
@@ -90,12 +91,11 @@ public class NativeExecutionSystemConfig
     private int numIoThreads = 30;
     private int shutdownOnsetSec = 10;
     private int systemMemoryGb = 10;
-    private DataSize queryMemoryGb = new DataSize(systemMemoryGb, DataSize.Unit.GIGABYTE);
+    // Reserve 2GB from system memory for system operations such as disk
```

Comment:
Should this be configurable and not hard-coded?

Reply:
Yeah, they are configurable like through tw in Meta internally. These are the defaults when we launch presto cpp if not configured. Thanks!